### PR TITLE
Add process for assigning milestones to issues/PRs

### PR DESCRIPTION
**Pull request by arhag**

### CODE REVIEW

File: process/work_tracking/milestones.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+This document captures the process of when and how to assign a milestone to an issue or pull request (PR).
+
+## Types of milestones
+
+Our repos typically have a milestone that tracks an upcoming version to be released of the product tracked by the repo. Some repos may have more granular milestones leading up to the next minor/major version of the product.
+
+The milestones can be divided into three categories:
+
+* Patch milestone: A patch milestone tracks a patch release of the product which is meant to only contain bug fixes and no new features. The patch milestone is typically named following a pattern where it has a prefix referring to the name of the product followed by a version number `X.Y.Z`. The `Z` number refers to the patch version number which is incremented as part of the new release. It is also possible for a patch milestone to track a release candidate version other than the initial `rc1` release; in this case the version would be something like `X.Y.0-rc2`.
```
Comment (ericpassmore) [inline process/work_tracking/milestones.md:11]: nitpick, what do you think about removing the the last sentence on release milestones, and simply referring to the 3rd and final bullet point.

File: process/work_tracking/milestones.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+If an issue tracks a bug fix in a release candidate or it includes other refinements to the release candidate prior to the stable release, e.g. documentation updates, then it should be assigned the appropriate release milestone. For example, consider work on top of a recently released Leap 5.0.0-rc1. If a bug was discovered in Leap 5.0.0-rc1 that requires a new release candidate to test, then the issue tracking the bug would be assigned the `Leap 5.0.0-rc1` milestone. There may also be some issues that are not considered bugs but must be completed prior to being able to release a stable version of 5.0.0; those issues would be assigned the `Leap 5.0.0` milestone.
```
Comment (ericpassmore) [inline process/work_tracking/milestones.md:23]: good point

**Review by ericpassmore — APPROVED** [review]
LGTM

File: process/work_tracking/milestones.md (hunk 1 above)
Comment (arhag) [reply process/work_tracking/milestones.md]: I realized that a release milestone is not necessary. In my latest commit I simplified the document a little bit by removing it.

Comment (ericpassmore) [reply process/work_tracking/milestones.md]: LGTM

File: process/work_tracking/milestones.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+If an issue is not a bug fix or changes for an existing release candidate, e.g. if it is an enhancement or new feature, then it should be assigned the appropriate feature milestone. For example, consider a scenario within the `leap` repo where the `main` branch is tracking work for an upcoming 5.0 release of Leap. A new issue tracking a new feature for 5.0 would likely be assigned the `Leap 5.0.0-rc1` feature milestone, unless there is an earlier and more granular feature milestone for it that is more appropriate.
+
+## Assigning milestones to PRs
+
+Pull requests typically should not have milestones attached to them. The only exception is for the PRs that forward bug fixes from a release branch forward to another release branch. In that case, the PR forwarding to a release branch should be assigned a patch milestone corresponding to the version of the target release branch.
```
Comment (spoonincode) [inline process/work_tracking/milestones.md:27]: The only uncertainty I have is whether PRs _without a linked issue_ also play by the same rules here: no milestone?

**Review by spoonincode — APPROVED** [review]

File: process/work_tracking/milestones.md (hunk 3 above)
Comment (arhag) [reply process/work_tracking/milestones.md]: I'm not sure. I left it ambiguous for now. I guess that is something we can figure out later as we further refine this process.

**Pull request merged**